### PR TITLE
Allow tcp:Caller as a parameter in onBytes() method

### PR DESCRIPTION
**Pull request by MohamedSabthar**

### CODE REVIEW

File: tcp-ballerina/tests/mock_server.bal [hunk 1]
```diff
@@ -32,20 +32,15 @@ service on echoServer {
...
 
 service class EchoService {
-    Caller caller;
 
-    public isolated function init(Caller c) {
-        self.caller = c;
-    }
-
-    remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|Error? {
+    remote function onBytes(Caller caller, readonly & byte[] data) returns (readonly & byte[])|Error? {
```
Comment (shafreenAnfar) [inline tcp-ballerina/tests/mock_server.bal:41]: In this case can't we change the signature as below ?

```ballerina
remote function onBytes(Caller caller, readonly & byte[] data) 
```

**Comment by shafreenAnfar** [comment]
```ballerina
remote function onBytes(Caller caller, readonly & byte[] data)
remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|error
```

Both of the above signatures are working right ?

**Comment by MohamedSabthar** [comment]
> ```ballerina
> remote function onBytes(Caller caller, readonly & byte[] data)
> remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|error
> ```
> 
> Both of the above signatures are working right ?

yes

File: tcp-ballerina/tests/mock_server.bal (hunk 1 above)
Comment (MohamedSabthar) [reply tcp-ballerina/tests/mock_server.bal:41]: we can

Comment (MohamedSabthar) [reply tcp-ballerina/tests/mock_server.bal:41]: ballerina```
    remote function onBytes(Caller caller, readonly & byte[] data) returns Error? {
        io:println("Echo: ", 'string:fromBytes(data));
        Error? res = caller->writeBytes(data);
    }
    ``` it's possible this way

Comment (MohamedSabthar) [reply tcp-ballerina/tests/mock_server.bal]: updated to `remote function onBytes(Caller caller, readonly & byte[] data) returns Error?`

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**